unit_tests/test_wait.py

```diff
@@ -80,6 +82,108 @@ def callback(arg1, arg2):
         self.assertEqual(len(calls), 1)
 
 
+from parameterized import parameterized
+
+
+class TestSdcmWaitWithEventStop(unittest.TestCase):
+
+    def setUp(self):
+        self.calls = []
+        self.callback_return_true_after = 0
+        self.ev = threading.Event()
+
+    def tearDown(self):
+        self.calls = []
+        self.callback_return_true_after = 0
+        self.ev.set()
+
+    def callback(self, arg1, arg2):
+        self.calls.append((arg1, arg2))
+        if len(self.calls) == self.callback_return_true_after:
+            return "what ever"
+        return False
+
+    def set_stop_in_timeout(self, ev: threading.Event, set_after: int):
+        while not ev.is_set():
+            if len(self.calls) == set_after:
+                ev.set()
+
+    @parameterized.expand([(True, ), (False, )])
+    def test_04_stop_by_event(self, throw_exc):
+        self.callback_return_true_after = 3
+        th = threading.Thread(target=self.set_stop_in_timeout, kwargs={"ev": self.ev, "set_after": 1})
+        th.start()
+        if throw_exc:
+            self.assertRaisesRegex(Exception, "callback: stopped by Event", wait_for,
```

Comment:
It must be some `ValueError` exception where we should see a message about the incompatibility of the `throw_exc` parameter and the `stop_event`. Isn't it?

Reply:
if throw_exc is true and wait_for is stopped by event, then Retry exception will be raised with appropriate error message. and test expect that , if false, then exception will not be raised